chain/epoch_manager/src/lib.rs

```diff
@@ -157,43 +161,55 @@ impl EpochManager {
             all_proposals, validator_kickout, validator_tracker, num_expected_blocks
         );
 
-        Ok((hash, new_proposals, all_proposals, validator_kickout, total_gas_used))
+        Ok(EpochSummary {
+            last_block_hash: hash,
+            all_proposals,
+            validator_kickout,
+            validator_online_ratio,
+            total_gas_used,
+        })
     }
 
     /// Finalizes epoch (T), where given last block hash is given, and returns next next epoch id (T + 2).
     fn finalize_epoch(
         &mut self,
         store_update: &mut StoreUpdate,
-        epoch_id: &EpochId,
+        block_info: &BlockInfo,
         last_block_hash: &CryptoHash,
         rng_seed: RngSeed,
     ) -> Result<EpochId, EpochError> {
-        let (
-            last_block_hash_prev_epoch,
-            new_proposals,
+        let EpochSummary {
```

Comment:
why unpack it right away, can just let epoch_summary and use epoch_summary.total_gas_used?

Reply:
Otherwise you have problems with partial moves.